orchagent/tunneldecaporch.cpp

```diff
@@ -249,44 +287,420 @@ void TunnelDecapOrch::doTask(Consumer& consumer)
+                        {
+                            src_ip_str = subnetDecapConfig.src_ip_v6->to_string();
+                        }
+                        else
+                        {
+                            SWSS_LOG_ERROR("%s: source IPv6 is not configured for subnet decap term, ignored.", key.c_str());
+                            it = consumer.m_toSync.erase(it);
+                            continue;
+                        }
+                    }
+                }
+                else if (is_subnet_decap_v4_term || is_subnet_decap_v6_term)
+                {
+                    SWSS_LOG_ERROR("%s: subnet decap is disabled, ignored.", key.c_str());
+                    it = consumer.m_toSync.erase(it);
+                    continue;
+                }
+
+                if (tunnel_exists)
+                {
+                    if (!addDecapTunnelTermEntry(tunnel_name, src_ip_str, dst_ip_str, term_type, subnet_type))
+                    {
+                        SWSS_LOG_ERROR("%s: failed to add tunnel decap term to ASIC_DB.", key.c_str());
+                    }
+                }
+                else
+                {
+                    SWSS_LOG_INFO("%s: tunnel doesn't exist, added to unhandled list.", key.c_str());
+                    addUnhandledDecapTunnelTerm(tunnel_name, src_ip_str, dst_ip_str, term_type, subnet_type);
+                }
+            }
+        }
+        else if (op == DEL_COMMAND)
+        {
+            if (tunnel_exists)
+            {
+                if (removeDecapTunnelTermEntry(tunnel_name, dst_ip_str))
+                {
+                    RemoveTunnelIfNotReferenced(tunnel_name);
+                }
+                else
+                {
+                    SWSS_LOG_ERROR("Failed to remove tunnel decap term %s from ASIC_DB.", key.c_str());
+                }
             }
             else
             {
-                SWSS_LOG_ERROR("Tunnel cannot be removed since it doesn't exist.");
+                SWSS_LOG_INFO("Tunnel for decap term %s doesn't exist, removed from unhandled list.", key.c_str());
+                removeUnhandledDecapTunnelTerm(tunnel_name, dst_ip_str);
             }
         }
+        else
+        {
+            SWSS_LOG_ERROR("Unknown operation type %s.", op.c_str());
+        }
+
+        it = consumer.m_toSync.erase(it);
+    }
+}
+
+void TunnelDecapOrch::doSubnetDecapTask(Consumer &consumer)
+{
+    SWSS_LOG_ENTER();
 
+    auto it = consumer.m_toSync.begin();
+    while (it != consumer.m_toSync.end())
+    {
+        KeyOpFieldsValuesTuple &t = it->second;
+        doSubnetDecapTask(t);
         it = consumer.m_toSync.erase(it);
     }
 }
 
+void TunnelDecapOrch::doSubnetDecapTask(const KeyOpFieldsValuesTuple &tuple)
+{
+    SWSS_LOG_ENTER();
+
+    string key = kfvKey(tuple);
+    string op = kfvOp(tuple);
+
+    bool valid = true;
+    bool set_src_ip = false;
+    bool set_src_ip_v6 = false;
+    string src_ip_str;
```

Comment:
Seems `set_src_ip` and `set_src_ip_v6` is not necessary. It can be replaced by `!src_ip_str.empty()` or `!src_ip_v6_str.empty()`